Update zenith and azimuth angle documentation (#2485)
pvlib/pvlib-python: docs/sphinx/source/user_guide/extras/nomenclature.rst

    
    aoi
        Angle of incidence. Angle between the surface normal vector and the

>         vector pointing towards the sun's center. Must be >=0 and <=180 degrees.

Reviewer: @OmarBahamida to let this PR move ahead without getting really involved, let's leave out any statements about a range of values.

    
    aoi
        Angle of incidence. Angle between the surface normal vector and the
        vector pointing towards the sun's center. [°]
    
    aoi_projection
        cos(aoi). When the sun is behind the surface, the value is negative.
        For many uses, negative values must be set to zero.

    ape
        Average photon energy

    apparent_zenith
        Apparent (refraction-corrected) solar zenith angle. The solar
        zenith angle describes the position of the sun relative to the vertical and is
        defined as the angle between the sun’s rays and a vector pointing straight up

>         from the observer. [°]

Reviewer: Nit: the angle is between a vector pointed straight up and a vector pointed at the sun, from the observer.

Reviewer 2: Applied this suggestion

        Diode saturation current

    solar_azimuth
        Azimuth angle of the sun in degrees East of North. The solar azimuth angle

>         describes the sun’s position along the horizon relative to a reference point.
>         The pvlib-python convention is defined as degrees east of north, so

Reviewer: ```suggestion
        describes the sun’s position along the horizon relative to the observer.
        The pvlib-python convention is defined as degrees East of North, so
```

        North = 0°, East = 90°, South = 180°, West = 270°.

    solar_zenith

>         Zenith angle of the sun in degrees. This is the angle between the sun's rays and

Reviewer: Nit: the angle is between a vector pointed straight up and a vector pointed at the sun, from the observer.

        is composed of direct and diffuse components.
    
    surface_azimuth
        Azimuth angle of the surface in degrees East of North. Must be >=0 and <=360.
        The convention is defined as degrees east (clockwise) of north. This is pvlib's

>         convention; other tools may use different conventions. For example, North = 0°,

Reviewer 3: I suggest moving the sentence, "For example, North...", before the sentence, "This is pvlib's convention...". 

As it is, I interpret the example as an example of some other tool's convention, not as an example of pvlib's convention.

        is composed of direct and diffuse components.
    
    surface_azimuth
        Azimuth angle of the surface in degrees East of North. This angle describes the

>         horizontal orientation of the PV panel relative to a reference direction. The

Reviewer: "horizontal orientation" seems vague. The azimuth angle is the azimuth of the surface normal vector, i.e., the vector pointed toward the sky perpendicular to the surface. That seems a bit wordy.

Reviewer 2: how about `horizontal normal vector from the array surface`? ([PVPMC](https://pvpmc.sandia.gov/modeling-guide/1-weather-design-inputs/plane-of-array-poa-irradiance/calculating-poa-irradiance/angle-of-incidence-clone/))

Reviewer: horizontal projection of the normal vector from the surface

>         pvlib-python convention is defined as degrees east (clockwise) of north, so

Reviewer: ```suggestion
        pvlib-python convention is defined as degrees East (clockwise) of North, so
```

>         North = 0°, East = 90°, South = 180°, West = 270°. Other tools may use different
>         conventions. 

Reviewer: ```suggestion
        North = 0°, East = 90°, South = 180°, West = 270°.
```

I don't think it's necessary to say other tools may be different.

    surface_tilt

>         Panel tilt from horizontal [°]. The surface tilt angle describes

Reviewer: ```suggestion
        Tilt from horizontal [°]. The surface tilt angle 
```

>         the inclination of the PV panel and is defined as degrees from the horizontal

Reviewer: ```suggestion
        is defined as degrees from the horizontal
```
